### ISSUE TITLE
SLIP-0010: Clarification required on conflicting instructions regarding the seed

### ISSUE THREAD

**Issue by Esya** [satoshilabs/slips#1120]
Currently [SLIP-0010](https://github.com/satoshilabs/slips/blob/master/slip-0010.md#master-key-generation) has conflicting instructions regarding the shape of the seed.

It states that it is "compatible to BIP-0032", implying that 128 or 256 bits seeds should be accepted : 
> It uses a process similar and compatible to BIP-0032.

It also has [128-bits seeds test vectors](https://github.com/satoshilabs/slips/blob/master/slip-0010.md#test-vector-1-for-ed25519) 

But the specification also clearly states : 

> 1. Generate a seed byte sequence S of 512 bits according to BIP-0039.

Could it be clarified, and stated clearly, one way or another if : 

- Valid SLIP-0010 implementations should accept seeds ranging from 128 to 512 bits, be them BIP-0039 seeds or BIP-0032 seeds, for backward compatibility - in that case amend the master key generation process to clarify this
- Valid SLIP-0010 implementations should reject seeds that are not coming from BIP-0039 seeds (In that case, maybe the 128 bits test vectors should be removed ?)

[Reference discussion that started this](https://github.com/oasisprotocol/oasis-sdk/pull/119#discussion_r631799623)

Thanks!

**Comment by prusnak** [comment]
You are confusing the mnemonic seed (encoded in words, which indeed has 128 or 256 bits) and the seed byte sequence S that is the output of PBKDF2-SHA512 which takes seed and the passphrase to produce the initial 512-bit entropy for BIP-0032.

Maybe @andrewkozlik can explain it better?

**Comment by Esya** [comment]
> You are confusing the mnemonic seed (encoded in words, which indeed has 128 or 256 bits) and the seed byte sequence S that is the output of PBKDF2-SHA512 which takes seed and the passphrase to produce the initial 512-bit entropy for BIP-0032.
> 
> Maybe @andrewkozlik can explain it better?

I'm talking about the seed byte sequence `S` in SLIP-0010 - not the mnemonic seed. Using BIP-0039, no matter what the length of the mnemonic seed is, we end up with a 512 bits S. My question is, why are there test vectors with 128-bits S then, and can we clarify the specs to address that ?

**Comment by andrewkozlik** [comment]
@Esya you are correct. Valid SLIP-0010 implementations should accept seeds ranging from 128 to 512 bits. I will go through the document and prepare a PR to clarify this.

**Comment by Esya** [comment]
Thank you for the clarification !

**Comment by andrewkozlik** [comment]
@Esya 
PR: https://github.com/satoshilabs/slips/pull/1122

**Comment by xianggithub01** [comment]
我来了
